src/python/pants/engine/legacy/graph.py

```diff
@@ -405,16 +425,12 @@ class LegacyHydratedTarget:
     """A rip on HydratedTarget for the purpose of V1, which must use BuildFileAddress rather than
     Address."""
 
-    address: BuildFileAddress
+    build_file_address: BuildFileAddress
     adaptor: TargetAdaptor
-    dependencies: Tuple[Address, ...]
-
-    def __hash__(self):
-        return hash(self.address)
```

Comment:
Unlike here, e want to hash on both the address and adaptor. This ensures that when targets change, but have the same address, that we don't end up with a hash bucket with too many similar entries.

There's no reason for us to implement our own `__hash__` and `__eq__` because the dataclass will generate the same implementations, unlike with `HydratedTarget` (because `HydratedTarget` has a private field, unlike here).